.github/workflows/release_chart.yaml

```diff
@@ -1,183 +1,183 @@
-          echo "Generating index.yaml within helm repository context..."
-          
-          # Copy the package to a local directory within helm repo
-          mkdir -p .cr-release-packages
-          cp "$WORK_DIR"/.cr-release-packages/*.tgz .cr-release-packages/ || {
-            echo "ERROR: No chart packages found in .cr-release-packages"
-            exit 1
-          }
-          
-          # Generate index with cr (this will handle the gh-pages branch automatically)
-          cr index \
-            --git-repo=${{ vars.HELM_GITHUB_REPOSITORY }} \
-            --owner=${GITHUB_REPOSITORY_OWNER} \
-            --release-name-template=${{ github.event.release.name }} \
-            --token=${{ secrets.HELM_GITHUB_TOKEN }} \
-            --package-path=.cr-release-packages \
-            --index-path=index.yaml \
-            --push || {
-            echo "ERROR: Failed to generate or push index to helm repository"
-            echo "Debug: Current directory is $(pwd)"
-            echo "Debug: Git remotes:"
-            git remote -v
-            echo "Debug: Git status:"
-            git status
-            exit 1
-          }
-          
-          echo "Successfully updated helm repository index"
-          
-          # Cleanup
-          cd /
-          rm -rf "$TEMP_DIR"
+name: release_chart
+
+on:
+  release:
+    types:
+    - published
+    - edited
+
+jobs:
+  release_chart:
+    name: Release Chart
+    permissions:
+      contents: write
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+
+      - name: Install chart-releaser
+        run: |
+          wget https://github.com/helm/chart-releaser/releases/download/v1.4.1/chart-releaser_1.4.1_linux_amd64.tar.gz
+          tar -zxf chart-releaser_1.4.1_linux_amd64.tar.gz cr
+          sudo install cr /usr/local/bin/
+          rm -f cr chart-releaser_1.4.1_linux_amd64.tar.gz
+
+      - name: Package Chart
+        run: cr package deploy/helm/clickhouse-operator
+
+      - name: Get Release Assets
+        id: get_assets
+        run: |
+          CHART_PATH=$(ls .cr-release-packages/altinity-clickhouse-operator-*.tgz)
+          ASSET_NAME=$(basename ${CHART_PATH})
+          ASSET_ID=$(curl -s -H "Authorization: Bearer ${{ secrets.GITHUB_TOKEN }}" \
+          "https://api.github.com/repos/${{ github.repository }}/releases/${{ github.event.release.id }}/assets" | \
+          jq -r ".[] | select(.name == \"$ASSET_NAME\") | .id")
+          
+          echo "Asset ID is $ASSET_ID"
+          echo "asset_id=$ASSET_ID" >> $GITHUB_OUTPUT
+
+      - name: Delete Existing Release Artifacts
+        if: steps.get_assets.outputs.asset_id != ''
+        run: |
+          curl -X DELETE -H "Authorization: Bearer ${{ secrets.GITHUB_TOKEN }}" \
+          "https://api.github.com/repos/${{ github.repository }}/releases/${{ github.event.release.id }}/assets/${{ steps.get_assets.outputs.asset_id }}"

```

Comment:
i really happy we found it

Reply:
Well, the level of this token access does not allow to delete repo, but thanks anyway